library/Rules/Json.php

```diff
@@ -15,11 +15,12 @@ class Json extends AbstractRule
 {
     public function validate($input)
     {
-        if (is_string($input)
-            && strtolower($input) == 'null') {
-            return true;
+        if (!is_string($input)) {
+            return false;
         }
 
-        return (null !== json_decode($input));
+        $decode = json_decode($input);
+
+        return is_object($decode) || is_array($decode);
```

Comment:
a few things I got from this (taking a quick reading...):

this [RFC 7159](https://tools.ietf.org/html/rfc7159) is an update of [RFC 4627](https://tools.ietf.org/html/rfc4627). in 4627 a json is considered either an object or an array.

in 7149 it says: "Note that certain previous specifications of JSON constrained a JSON text to be an object or an array.". in the [examples section](https://tools.ietf.org/html/rfc7159#page-12) of 7149 it considers the following as three small JSON texts containing only values:

   * "Hello world!"
   * 42
   * true

using some json parsers on the web such as: http://jsonlint.com/, http://json.parser.online.fr/
they consider the three values above as invalid, I think they follow the 4627.

well, this changed my mind about only validating either an object or an array :)

what do we do now? :stuck_out_tongue_winking_eye: